File: maven-confluence-reporting-plugin/src/test/java/org/bsc/maven/plugin/test/FileIOTest.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package org.bsc.maven.plugin.test;

import org.apache.commons.io.FileUtils;
import org.hamcrest.core.Is;
import org.junit.Assert;
import org.junit.Test;

/**
 *
 * @author bsorrentino
 */
public class FileIOTest {
    
    @Test
    public void forceMkdir() throws Exception  {
        
        FileUtils.deleteDirectory( new java.io.File("target"));
        
        java.io.File outputFile = new java.io.File( "target/io", "test.txt");
        
        java.io.File folder = new java.io.File(outputFile.getParent());

        Assert.assertThat( folder.exists() , Is.is(false));
        
        FileUtils.forceMkdir( folder );

        Assert.assertThat( folder.exists() , Is.is(true));
        Assert.assertThat( folder.isDirectory() , Is.is(true));

        FileUtils.writeStringToFile(outputFile, "this is test");
        
        Assert.assertThat( outputFile.exists() , Is.is(true));
        Assert.assertThat( outputFile.isFile() , Is.is(true));
    }
}

File: mojo/src/test/kotlin/org/bsc/reporting/FileIOTest.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package org.bsc.reporting

import org.apache.commons.io.FileUtils
import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import java.io.File
import java.nio.charset.Charset

/**
 *
 * @author bsorrentino
 */
class FileIOTest {
    @Test
    @Throws(Exception::class)
    fun forceMkdir() {
        FileUtils.deleteDirectory(File("target/test"))

        val outputFile = File("target/test/io", "test.txt")
        val folder = File(outputFile.parent)

        assertFalse(folder.exists())

        FileUtils.forceMkdir(folder)

        assertTrue(folder.exists())
        assertTrue(folder.isDirectory)

        FileUtils.writeStringToFile(outputFile, "this is test", Charset.defaultCharset(), false)
        assertTrue(outputFile.exists())
        assertTrue(outputFile.isFile)
    }
}